### PR TITLE
Fix misusage of outDir vs. rootDir vs. baseDir.

### PR DESCRIPTION
**Pull request by dgoldstein0**
currently dts-generator works fine when the outDir & rootDir aren't set, so that the outputs and inputs are collocated.  When that is not the case, assuming outputs end up in baseDir instead of outDir produces incorrect module names.  Additionally, we need to preserve the rootDir option to the compiler for any projects that use it, as that also impacts output destinations and indirectly our generated module names.

This diff and #92 point out that copying `compilerOptions` from the tsconfig.json to `options`, and then from `options` to the new `compilerOptions` is error prone and buggy; I'm thinking of cleaning this up in a future pull request, so we can fighting this class of bugs.

### CODE REVIEW

File: tests/support/foo-directories/sub/Bar.tsx [hunk 1]
```diff
@@ -0,0 +1,8 @@
+export default class Bar {
```
Comment (dylans) [inline tests/support/foo-directories/sub/Bar.tsx]: Is there anything specifically tsx about this example?

File: tests/support/foo-directories/tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,17 @@
+{
+	"version": "1.6.2",
+	"compilerOptions": {
+		"target": "es5",
+		"module": "umd",
+		"jsx": "react",
```
Comment (dylans) [inline tests/support/foo-directories/tsconfig.json]: I have no problem with adding an example that specifically tests the React/JSX/TSX approach, but it seems like this example may be conflating the directory issue and adding in some tsx specific details?

File: tests/unit/index.ts [hunk 3]
```diff
@@ -60,6 +60,31 @@ registerSuite({
...
+		// respect the rootDir option.  This project also has an outDir so this
+		// stresses our path-handling logic - if we mix up the directories, it'll
+		// show in the output module names.
+		//
+		// This project also includes a tsx file and uses absolute paths, for extra
+		// fun.
+		return generate({
+			project: 'tests/support/foo-directories',
+			out: 'tmp/foo.config.d.ts',
+			main: 'tests/support/foo-directories/sub/index.tsx'
```
Comment (dylans) [inline tests/unit/index.ts]: Same question about tsx... maybe I'm missing something, does this issue only occur with tsx?

**Comment by dgoldstein0** [comment]
Yeah sorry, don't think the tsx is necessary here - I started by guessing
it was a tsx bug and when I finally found out otherwise, never removed that
bit.

**Pull request merged**